xml_converter/src/xml_converter.cpp

```diff
@@ -217,80 +249,72 @@ void process_data(
 }
 
 ////////////////////////////////////////////////////////////////////////////////
-// main
-//
-// Main is the CLI entrypoint to the xml converter. It handles processing all
-// of the command line data into a format the internal functions want to
-// receive.
+// parse_arguments
+// Processes all of the command line data into a format the internal functions
+// want to receive.
 //
 // Example usage
-//   ./xml_converter --input-taco-paths ../packs/marker_pack --output-split-guildpoint-path ../output_packs
-//   ./xml_converter --input-taco-paths ../packs/* --output-split-guildpoint-path ../output_packs
+//   ./xml_converter --input-taco-paths ../packs/marker_pack --output-guildpoint-path ../output_packs
+//   ./xml_converter --input-taco-paths ../packs/* --output-guildpoint-path ../output_packs --split_by_map_id
+//
+// --import_FORMAT_paths <path> --FLAGS [--import_FORMAT_paths <path> --FLAGS]
+// --export_FORMAT_paths <path> --filter "conditions" --FLAGS
+// [--export_FORMAT_paths <path> --filter "conditions" --FLAGS]
 ////////////////////////////////////////////////////////////////////////////////
-int main(int argc, char* argv[]) {
-    vector<string> input_taco_paths;
-    vector<string> output_taco_paths;
-    vector<string> input_guildpoint_paths;
-    vector<string> output_guildpoint_paths;
-    bool allow_duplicates = false;
-
-    // Typically "~/.local/share/godot/app_userdata/Burrito/protobins" for
-    // converting from xml markerpacks to internal protobuf files.
-    vector<string> output_split_guildpoint_paths;
-
-    vector<string>* arg_target = nullptr;
+vector<PathConfig> parse_arguments(int argc, char* argv[]) {
+    vector<PathConfig> path_configs;
+    PathConfig* arg_target = nullptr;
 
     for (int i = 1; i < argc; i++) {
         if (!strcmp(argv[i], "--input-taco-path")) {
-            arg_target = &input_taco_paths;
+            path_configs.emplace_back("import", "xml");
```

Comment:
Why did you choose `emplace_back` here?

Reply:
What I wanted to happen here in each `if` statement was to create a `PathConfig` object, populate it with the information provided by the argument so it could be sorted later, and use a reference to that object when looking at the next argument to see if it modifies the object further. Since the last part of that needs the correct location of the object, `&path_configs.back()` was used to get a pointer to the location on the vector. So for the other 2 steps, I found that using emplace creates the object directly into the vector as opposed to using push which would create an object and then move it into the vector. Both would achieve the same effect. Emplace then had the pro of being one line of code versus four if using push_back.

Reply:
That is reasonable.